src/ui/MessageHeader/index.tsx

```diff
@@ -0,0 +1,44 @@
+import React, {ReactElement} from 'react';
+import './index.scss';
+import Label, {LabelColors, LabelTypography} from '../Label';
+import {CoreMessageType, getSenderName, SendableMessageType} from '../../utils';
+import {Nullable} from '../../types';
+import {GroupChannel, Member} from '@sendbird/chat/groupChannel';
+
+export interface MessageHeaderPublicProps {
+  text: string;
+}
+
+export interface MessageHeaderInternalProps {
+  channel: Nullable<GroupChannel>;
+  message: CoreMessageType;
+}
```

Comment:
How about just get the text?

```tsx
<MessageHeader text={'... typings'} />
<MessageHeader text={message.sender.nickname} />
```

Reply:
Good idea but unfortunately then we can't do
```
renderMessageHeader={(props) =>{
  <MessageHeader {...props}/>
}}
```

Reply:
You can do it in here
```
        {
          !isByMe && !chainTop && !useReplying && renderMessageHeader({ ...props, text: getHeaderText(channel, message) })
        }
```

Reply:
oh thats good idea but do we really want to define interface to be two ways? We MUST make text as optional but we use it like it is required in the implementation.
```
export interface MessageHeaderProps {
  channel: Nullable<GroupChannel>,
  message: CoreMessageType,
  text?: string;
}

export default function MessageHeader(props: MessageHeaderProps): ReactElement {
  return (
    <Label
      className="sendbird-message-content__middle__sender-name"
      type={LabelTypography.CAPTION_2}
      color={LabelColors.ONBACKGROUND_2}
    >
      { props.text }
    </Label>
  );
}
```